frontend/src/App/App.jsx

```diff
@@ -37,6 +37,25 @@ class App extends React.Component {
     };
   }
 
+  setupMetadata = () => {
+    const paths = window.location.pathname?.split('/').filter((path) => path !== '');
+    const subpath = window.public_config?.SUB_PATH;
+    const subpath_length = subpath ? subpath.split('/').filter((path) => path !== '').length : 0;
+
+    if (paths.length - subpath_length === 2 && paths[subpath_length] === 'applications') {
+      // call apps api and get the app type
+      appService.getAppBySlug(paths[length]).then((app) => {
+        if (app && !app.is_public) {
+          if (this.state.currentUser) {
+            this.fetchMetadata();
+          }
+        }
+      });
+    } else {
+      this.fetchMetadata();
```

Comment:
Shall we skip the metadata API call if route includes `applications` and add fetch metadata call in app viewer page?
- Advantage of this: we have the app details in app viewer page and can invoke fetch meta call if application is not public

Reply:
ok, got it